subcommand/mesh-init/command.go

```diff
@@ -25,15 +27,25 @@ type Command struct {
 	config *config.Config
 	once   sync.Once
 	log    hclog.Logger
+
+	flagSet *flag.FlagSet
+	logging.LogOpts
 }
 
 func (c *Command) init() {
-	c.log = hclog.New(nil)
+	c.flagSet = flag.NewFlagSet("", flag.ContinueOnError)
+	logging.Merge(c.flagSet, c.LogOpts.Flags())
 }
 
-func (c *Command) Run(_ []string) int {
+func (c *Command) Run(args []string) int {
 	c.once.Do(c.init)
 
+	if err := c.flagSet.Parse(args); err != nil {
+		return 1
+	}
```

Comment:
We use JSON config instead of flags for `mesh-init` and `health-sync`, but we don't support JSON config for all commands (`acl-controller`, `app-entrypoint`, `envoy-entrypoint`).

Should we:

1. Use a `-log-level` flag for all commands?
2. Use a JSON `logLevel` field for `mesh-init` and `health-sync`, and `-log-level` for other commands?
3. Switch to JSON for all commands, and use a JSON `logLevel` field for all commands? (This would be a breaking change though at this point).

Reply:
I think 2. mesh-init and health-sync are supposed to be user facing which is why we switched to json. But acl controller isn't user-facing and the entrypoints would probably be annoying to use if you had to write json.